### DISCUSSION TITLE
macOS Support Group

### DISCUSSION THREAD

**Discussion by mtdvlpr** [sircharlo/meeting-media-manager#3222] [General]
# macOS Support Group

## Why?

Dear users, I have tagged some of you here, because you've been linked in one way or another with M³ issues involving macOS.

The reason for this, is that neither @sircharlo nor I use M³ on macOS and I think it would be helpful to have a place where we can ask you for assistance with testing macOS specific features. If you don't want to be included in this, feel free to unsubscribe from this discussion. If you do want to be included, thank you for your willingness to help!

## What?

What I expect from this discussion is that every once in a while @sircharlo or I will post a question or request here in the hopes that some of you are available to help. If you have a macOS specific issue, problem, request or question, please use the designated channels for that (report an issue or open a discussion). That way, this discussion can focus on the questions and requests we actually need your help with.

## Where?

I thought this would be a good place for it, but if you guys have a better idea of how we could set this up, please let me know!

> [!IMPORTANT]
> The questions and requests might involve installing a test version of M³. If this is the case, **don't** do this in the Kingdom Hall! Test versions could be unstable.

@yarko-w, @ux-git, @skimark1, @Jiang-TJ, @alanfermtz, @vangquan, @sbscomp, @Danreynolds124

**Comment by mtdvlpr** [comment]
## Website Presentation Bug

Please take a look at #3217 and see if this issue also applies to you and if it does whether the latest dev build fixes it (see answer in issue for latest dev build). Also take a look at #3190 and see if the same dev build correctly fixes that issue as well.

Please share your results in #3217.

**Comment by ux-git** [comment]
Just tested the latest version vs the exact same one (downloaded from GitHub, not through Wetransfer though, so I'm not 100% sure). I gave all the permissions, including multiple attempts of giving the screen record permission.

Neither present the jw.org unfortunately, can't get why and HOW, considering that I definitely did take [a screenshot of the second display](https://github.com/sircharlo/meeting-media-manager/issues/2768#issuecomment-2457443449) while presenting.

Totally weird. The apps quits well on the latest version though.

**Comment by mtdvlpr** [comment]
## Website Presentation Bug 

Please take a look at #3217. Is the present website feature still broken in `v24.11.1`? I want to pinpoint exactly where the feature started breaking. Could you guys verify whether the present website works in each of the old test versions below?

### Installing test versions

If you run into problems installing any test version, please [follow the steps defined in our documentation website](https://sircharlo.github.io/meeting-media-manager/using-at-a-kingdom-hall#additional-steps-for-macos-users).

### Verifying functionality

To verify whether the feature works or not, perform the following steps:

<details>
<summary>Step 1: Open the "Website presentation" page</summary>

![image](https://github.com/user-attachments/assets/45ac9519-2bf0-4636-a2db-8c9a6b879664)

</details>

<details>
<summary>Step 2: Press the "Start mirroring" button</summary>

![image](https://github.com/user-attachments/assets/607bba1f-fcc0-4e5a-bf10-acfb359c6d20)

</details>

<details>

<summary>
Step 3: Verify whether 1) the window that's created shows the official website and 2) the media window shows a mirror of the window that's created
</summary>

![image](https://github.com/user-attachments/assets/19f1bb25-1604-4d2c-a531-f433db6a7d39)

</details>

### Test Versions

#&#8203;24.11.1 Latest official release:
https://github.com/sircharlo/meeting-media-manager/releases/tag/v24.11.1

**Reply by ux-git** [reply]
I just ran through each of the test versions mentioned – even with deep uninstalling with Clean My Mac app, to clear all the caches –  but none of the versions worked on my M-silicone Mac. I'll try once again on an Intel one later.

**Reply by mtdvlpr** [reply]
Thank you for your help!

**Comment by skimark1** [comment]
I’m not completely sure what I’m looking for, but on my M2 Mac and Intel Mac I don’t have any erratic behavior. Could you specify the bug?

Thanks,
Mark Cillis
***@***.***

**Reply by sircharlo** [reply]
The Website Presentation feature isn't working for some. Is it working for you?

**Reply by mtdvlpr** [reply]
@skimark1, did you follow below steps? The bug is that the media window doesn't show the website, but instead a blank, black screen. Please let us know if you see a blank, black screen or thr actual jw.org website.

### Verifying functionality

To verify whether the feature works or not, perform the following steps:

<details>
<summary>Step 1: Open the "Website presentation" page</summary>

![image](https://github.com/user-attachments/assets/45ac9519-2bf0-4636-a2db-8c9a6b879664)

</details>

<details>
<summary>Step 2: Press the "Start mirroring" button</summary>

![image](https://github.com/user-attachments/assets/607bba1f-fcc0-4e5a-bf10-acfb359c6d20)

</details>

<details>

<summary>
Step 3: Verify whether 1) the window that's created shows the official website and 2) the media window shows a mirror of the window that's created
</summary>

![image](https://github.com/user-attachments/assets/19f1bb25-1604-4d2c-a531-f433db6a7d39)

</details>

**Comment by skimark1** [comment]
When I start mirroring, I get a black screen and the official website as a separate window. It is the same in all versions and on both Macs.

Thanks,
Mark Cillis
***@***.***

**Reply by mtdvlpr** [reply]
Alright. Thank you for testing! That black screen is supposed to also show the official website. @sircharlo, I'm afraid we'll have to disable this feature again for macos. It seems that the one time we got it working was just a fluke...

**Comment by skimark1** [comment]
Before you disable it, I found something out. The first time I try to mirror, I get the request to use mic and camera. I accepted them and had the 2 windows. One with jw.org <http://jw.org/> and the other a black one. I stopped mirroring, and clicked to mirror again, and got another request to bypass system settings to use mic and camera. I granted access to it and now I have a small window of jw.org <http://jw.org/> and the black window is a a mirror of the small window. This looks like it is working correctly, just an extra step to allow it. Will test on M2, this was first confirmed on Intel Mac

Thanks,
Mark Cillis
***@***.***

**Reply by mtdvlpr** [reply]
Awesome! What version of M3 did you use?

**Reply by mtdvlpr** [reply]
@sircharlo, the 3rd permission not being asked the first time might be a race condition. I think if we ask the mic and camera permissions on present website page load rather than on the mirror action, those permissions will be handled in time for the 3rd permission to be prompted on the mirror action.

**Reply by mtdvlpr** [reply]
> @sircharlo, the 3rd permission not being asked the first time might be a race condition. I think if we ask the mic and camera permissions on present website page load rather than on the mirror action, those permissions will be handled in time for the 3rd permission to be prompted on the mirror action.

Implemented in 4c2f647dcc10a8e73e2152605aee5a932da908f1

**Comment by skimark1** [comment]
I just tested on M2 Mac and got almost identical results. The end result is that it is working! The window asking to bypass the system setting in M2 environment is hiding behind M3 and it is easy to miss it. Once I accepted it, everything started working fine.

Thanks,
Mark Cillis
***@***.***

**Reply by mtdvlpr** [reply]
Ah, that makes sense. The window that appears to browse the website is configured to always be on top.

**Reply by sircharlo** [reply]
I wonder if this is the issue; that the permissions window is hidden by the website? If so, maybe the website window should not be always on top to avoid this?

**Reply by ux-git** [reply]
@sircharlo this window is still always re-prompts the permission even thought they're given all the time. During each website mirroring I drag the main mirroring window away, grant the permissions (well, not really because I only get redirected to the settings page, where all the permissions are already given) and then on the next mirroring session, even if 10 secs later, it pops-up once again.

**Reply by ux-git** [reply]
@skimark1 I also experienced this exactly the way you describe it. But after trying to mirror the jw.org after restarting/updating the app, it stopped working. Can you confirm that the mirroring on the second display works after the app reboot, and perhaps later in a few days after update?

**Reply by mtdvlpr** [reply]
> I wonder if this is the issue; that the permissions window is hidden by the website? If so, maybe the website window should not be always on top to avoid this?

Implemented in 8855d09382f1ac6a6cf5b594efb4acd5a6fb4b19

**Comment by skimark1** [comment]
24.11.1 on both Macs, M2 and Intel.

Thanks,
Mark Cillis
***@***.***

**Reply by mtdvlpr** [reply]
Great! Are the permissions persisted now? Do you need to allow it once or everytime you use the app?

**Comment by skimark1** [comment]
I never experienced the continuous request for permissions. Just had to accept the first 2, separate from each other, and then after that came the one to bypass the system settings. Once I accepted that one, it is persistent.

Thanks,
Mark Cillis
***@***.***

**Reply by mtdvlpr** [reply]
Perfect. Thank you so much for this!

**Comment by skimark1** [comment]
I’m happy to help. Thanks for such a great tool!

Thanks,
Mark Cillis
***@***.***

**Comment by ux-git** [comment]
@mtdvlpr @sircharlo Here we go!

<img width="1917" alt="Screenshot 2024-11-11 at 20 39 53" src="https://github.com/user-attachments/assets/ca1b2495-a968-4e68-8a32-1fa72fd9a540">


Long story short: What helped me out was 1. uninstalling the app, 2. withdrawing the screen recording permissions (not just toggle off, but remove the M³ completely from the list), 3. install the latest version (I did a deep uninstall, but the app still somehow managed to restore my pre-redesign-verison data from my old congregation), 4. started mirroring, granted both new camera/mic permissions, and only after doing so 5. granted the screen recording permissions once again + quit and reopen.

Now rebooting the computer, restarting the app won't break the functionality... at least so far.

**Reply by mtdvlpr** [reply]
That's great news! I'm hoping to have fixed the need to quit an reopen for all permissions to be set, but if not, it's not the end of the world. Also, am I to understand from the screenshots that the black bars on the sides are gone and the full window is utilized?

**Reply by mtdvlpr** [reply]
@sircharlo, is this worth mentioning in the docs as well or do we wait for others to report issues to reference to this solution?

**Reply by sircharlo** [reply]
> @sircharlo, is this worth mentioning in the docs as well or do we wait for others to report issues to reference to this solution?

I'd say wait and see?

**Reply by ux-git** [reply]
@mtdvlpr the black bars are there, with the top window title also present.

My guess it that it was some kind of version conflict bug, when the permission given to the older version didn't really work for the new one, so I had to re-grant them. Just a guess

**Reply by mtdvlpr** [reply]
Could you demonstrate with a side-by-side comparison between the website controller window and the media window?

**Reply by ux-git** [reply]
The screenshot above actually demonstrates it well. The right monitor has the full-screen browser in the background, meaning everything on the left is the mirroring (with black bars and title)

**Comment by yarko-w** [comment]
<img width="472" alt="Screenshot 2024-11-12 at 7 09 53 am" src="https://github.com/user-attachments/assets/6e815bef-4ecb-433f-aa64-c735c810cdb2">

Hi Friends 👋 I installed the latest build yesterday and wasn't able to get it working. After reading through some of these comments I discovered the 3rd permission request window pictured was hidden behind the other windows. However in system settings it was supposedly  already granted. I followed @ux-git and manually removed the permission and added it again. Doing this automatically prompts it to quit and reopen and it has started working since. The first time it worked I also got the 'Allow for one month'  so will be good to review this again in a month and see if it breaks again.

**Reply by mtdvlpr** [reply]
Thanks! This is very helpful. In the next release, the permission request should not be hidden behind the window anymore.

**Reply by yarko-w** [reply]
Awesome! Also wondering if it would be helpful to remove the window frame (title bar) around the share window since it doesn't have controls enabled anyways and it is shown to the media window. Not sure if this is possible or helpful?

**Reply by mtdvlpr** [reply]
I was under the assumption the window frame (title bar) was already removed from the share window. This is the behavior on Windows and should (according to Electron's documentation) also work for macOS. We'll look into it!

**Reply by sircharlo** [reply]
Oh, macOS... 😆

**Reply by mtdvlpr** [reply]
See #3256

**Comment by skimark1** [comment]
I’ve closed and relaunched several times after waiting 3 hours and everything is still working as expected on M2 and Intel
.
Thanks,
Mark Cillis
***@***.***

**Reply by ux-git** [reply]
Perfect, thanks

**Comment by skimark1** [comment]
Just noticed that when I Quit M3, the main window closes, I get the warning to make sure it isn’t in use, and then quit again, but the icon stays in the dock showing it as running even though I’ve told it to quit. Happens with M2 and Intel both. You have to go to the icon, right click it and select quit again.

Thanks,
Mark Cillis
***@***.***

**Reply by ux-git** [reply]
The issue is still present after using Cmd+Q?

**Comment by skimark1** [comment]
You have to use CMD+Q 3 times to finally make it quit. When using the mirroring function, the first time gets the warning to make sure it isn’t still in use, second time closes the M3 window, but M3 is still shown as running in the dock. Third time makes it finally quit.

Thanks,
Mark Cillis
***@***.***

**Reply by ux-git** [reply]
The same on my end

**Reply by mtdvlpr** [reply]
See #3319

**Comment by wh575** [comment]
Hi friends, I've been using MMM on an M1 Mac for sometime with no issues. However, after installing v24.11.1 I can't open the app. As normal with Macos, I first need to authorize the app in settings by selecting "open anyway" where I'd normally need to enter my password or fingerprint. But at that point it crashes. Now this happens for previous versions I try install as well. Maybe I should do a deep uninstall?

**Reply by mtdvlpr** [reply]
Have your tried v24.11.2 which has just been released?

**Reply by wh575** [reply]
WOW quick reply! Yes, I did. I even rolled back to v24.11.0

**Reply by sircharlo** [reply]
Did you do the steps from https://sircharlo.github.io/meeting-media-manager/using-at-a-kingdom-hall#additional-steps-for-macos-users?

**Reply by wh575** [reply]
That was the reason. Problem solved! Thanks so much!

**Comment by skimark1** [comment]
I went from it working perfectly in 24.11.1 to now when I install and try to run 24.11.2, and running the scripts in the terminal, I get the request to allow permissions every time and nothing is displayed in the full screen window. Any ideas?

Thanks,
Mark Cillis
***@***.***

**Reply by mtdvlpr** [reply]
Did you follow the steps described below?

https://sircharlo.github.io/meeting-media-manager/using-at-a-kingdom-hall#screen-sharing-issues

**Comment by skimark1** [comment]
I took multiple attempts but here is what I did: 
Uninstall 24.11.2
Remove the M3 instance from the Privacy window
Apply the terminal commands
Reinstall 24.11.2
Launch and try to mirror screen, it prompted for permissions, and didn’t work after I allowed them, but after quitting and trying again, it worked the second time.

Thanks,
Mark Cillis
***@***.***

**Comment by skimark1** [comment]
Is this something that will possibly be overcome in the future? This will be a real challenge for the other 2 congregations in our hall, since they don’t have anyone that will understand this workaround.

Thanks,
Mark Cillis
***@***.***

**Reply by mtdvlpr** [reply]
Probably not. See #3259 for more details.

**Reply by sbscomp** [reply]
What I did, since I'm the security coordinator for our hall, is to set up a script which executes the requisite signing command, and set that up so that any user can execute it via sudo. So my instructions to the other congregations is that all they need to do is open a terminal and type "sudo /usr/local/bin/fix" and hit tab to auto-complete the script name ("fix_media_manager_permissions" is what I called it I think) and hit enter, and it'll fix the permissions. It's a pain but at least it's easy enough to do. You could probably create a script that also executed the sudo part and then put a shortcut to that on the desktop.

**Reply by sircharlo** [reply]
At the risk of sounding callous, how often do we really need to mirror the website during meetings?

**Reply by ux-git** [reply]
@sbscomp theoretically the script can even be placed at the desktop or in the dock for faster and more convenient access: 
 https://apple.stackexchange.com/questions/419767/how-to-create-a-shortcut-for-a-command-in-terminal-that-i-can-have-in-my-dock

**Comment by skimark1** [comment]
I just noticed the new version 24.11.4, and wondered, has the auto check/update feature been removed? Is the only path to update to download and install manually?


Thanks,
Mark Cillis
***@***.***

**Reply by sircharlo** [reply]
Unfortunately, macOS does not allow apps like this one to auto-update automatically.

**Reply by mtdvlpr** [reply]
We did add a banner for macOS that should appear in the app when a new update is available. Have you noticed that banner?

**Comment by skimark1** [comment]
I was running 24.11.2 and didn’t see a banner. Maybe added to a newer version?


Thanks,
Mark Cillis
***@***.***

**Reply by mtdvlpr** [reply]
The banner was added in 24.11.1. I'll double-check this behavior!

**Reply by mtdvlpr** [reply]
Could you try out this version? You should see an announcement banner indicating a new version is available.

https://github.com/sircharlo/meeting-media-manager/actions/runs/11912198441

**Comment by skimark1** [comment]
It’s’ been running for 10 minutes, still no banner. Where should it appear? I would think where the other banners are at the top of the window?


Thanks,
Mark Cillis
***@***.***

**Reply by mtdvlpr** [reply]
Hmm, very strange. How about this one?

https://github.com/sircharlo/meeting-media-manager/actions/runs/11916514976

You should see the following:

![image](https://github.com/user-attachments/assets/d38d375a-b534-4047-a357-4a08c0eaede7)

**Comment by skimark1** [comment]
The banner is showing up on Intel and M2 platforms. I get a message that an update has been downloaded, but nothing is in the downloads folder. When I click on the banner, I get directed to the download site. Is that expected behavior? Also, in M series Mac, you still have to Cmd/Quit 2-3 times to get it to actually quit, otherwise it keeps running even tho no windows show.


Thanks,
Mark Cillis
***@***.***

**Reply by mtdvlpr** [reply]
Alright, thanks! Yeah, the auto updater doesn't work for macOS. That's why the banner takes you to the download site. What does the banner say the platform is on intel / m2?

**Comment by skimark1** [comment]
The Cmd/Quit is also acting the same way on Intel. Have to Cmd/Quit at leas 2x to get it to quit. If you’re using the jw.org <http://jw.org/> feature, it takes 3x.


Thanks,
Mark Cillis
***@***.***

**Comment by skimark1** [comment]
They both say the platform is win

￼

**Reply by mtdvlpr** [reply]
Okay, thanks for testing! I've narrowed down the issue. You can install the latest official release again.

**Comment by skimark1** [comment]
I just updated to 24.11.5 on Intel Mac and there is something very wrong with the build. It is unusable. See attached video.

￼

Thanks,
Mark Cillis
***@***.***

**Reply by mtdvlpr** [reply]
Hi @skimark1, the video was not attached correctly. Could you try opening GitHub directly and uploading the video? Or uploading the video to a cloud service and providing us with the link?

**Comment by skimark1** [comment]
Also, there was no banner stating that there was a new version.

Thanks,
Mark Cillis
***@***.***

**Reply by mtdvlpr** [reply]
~The banner has been fixed, but it won't be shown until you have v24.12.0 and there's a newer update (24.12.1 or 25.1.0) available.~

**Reply by mtdvlpr** [reply]
Correction: it should be shown from v24.11.5 onward. So for thr next release you should see a banner.

**Reply by mtdvlpr** [reply]
@skimark1, do you see the banner now that there's a new version (v24.12.0)?

**Comment by skimark1** [comment]
Here's the video. As soon as you launch M3 it starts jumping on both screens and you almost can't quit it. Even trying to force quit is very difficult since the menu isn't stable. This is on Intel Mac, didn't want to test it on M4

https://github.com/user-attachments/assets/337ba033-095b-4f76-a336-685b78944bc4

**Reply by mtdvlpr** [reply]
Thanks for the video! @sircharlo, seems like it is stuck in a fullscreen/window toggling loop...

**Reply by mtdvlpr** [reply]
@skimark1, could you try the latest build? It has some improvements to the screen management.

https://github.com/sircharlo/meeting-media-manager/actions/runs/12106614960

**Reply by skimark1** [reply]
That stopped the screen jump, but when M3 is launched it starts playing background music automatically. Also, when quitting, it still takes to 'quits' to make the app fully close. The first quit makes the window disappear, but the app is still running in the dock. You have to right click the app or click on it to make it the active app so you can fully quit it.

**Reply by mtdvlpr** [reply]
Playing background music automatically is intended behavior when you have that setting turned on and the meeting starts in less than an hour. The two quits to quit is also intended behavior I believe. See https://github.com/sircharlo/meeting-media-manager/issues/3319#issuecomment-2478802025 for more details.

**Comment by skimark1** [comment]
The banner worked on Intel Mac and M2 Mac also. Everything working as expected. We used it for the meeting Tuesday for the first time and a couple of observations:

1. When the main window is moved, it causes the display window to flash and flicker
2. Suggestion: could a ’time remaining’ feature be added to the scrub bar on the media playback? This is helpful to the operator to know exactly how much time is remaining.

We appreciate your great work on this amazing tool!

Thanks,
Mark Cillis
***@***.***

**Reply by sircharlo** [reply]
Regarding point 1, is that with version 24.12.0?
For point 2, good idea. I'll add that.

**Comment by skimark1** [comment]
No, the flicker was with 24.11.6
Thanks,
Mark Cillis
***@***.***

**Reply by sircharlo** [reply]
So that is resolved in 24.12.0, correct?

**Comment by skimark1** [comment]
I won’t know until Sunday
Thanks,
Mark Cillis
***@***.***

**Comment by skimark1** [comment]
Hi,

The screen flicker is completely resolved! I would like to suggest that you add an option to keep a custom background at all times if desired. We have a custom background that compliments our decor of the Hall. We do use it every meeting and have to apply it manually. Having the option to retain would be appreciated by us and maybe others?

Also, having to quit M3 at least 2 times and sometimes 3 times is not normal behavior on a Mac. It seems that once you select quit, it should quit on the first request. If a newbie doesn’t see that it is still running, it is stealing resources from the system because still running.

We appreciate this great tool and thank you for your work on it and the continuing improvements.
Thanks,
Mark Cillis
***@***.***

**Comment by mtdvlpr** [comment]
# Exciting news!

Dear macOS users, we have some exciting news! Thanks to the generous help of @sbscomp, we are working on supporting signed and notarized macOS releases, and we have released the first signed and notarized beta version:

https://github.com/sircharlo/meeting-media-manager/releases

Please try different ways of installing it:

- Install the test version while the current version is installed
- Install the test version after uninstalling the current version
- Perform a completely clean install of the test version (remove all permissions that were given before)

This release should come with some major benefits:

- No more Terminal commands
--Persisted permissions between versions
- Auto update support

Please let us know your findings, and thank you in advance for your help!

**Reply by ux-git** [reply]
Thanks for the improvement the workaround for mac users! 

My feedback: Installing this new build over the last stable version still won't let me easily open it:
<img width="372" alt="Screenshot 2024-12-19 at 22 33 50" src="https://github.com/user-attachments/assets/901e10ee-1eff-49e0-a205-787864229b58" />

Double checked – after clean install it still shows this message.

**Reply by ux-git** [reply]
By the way, what do you think about making the beta version installable alongside the stable version? This approach would not require uninstalling or overwriting the currently functioning version and would allow for more seamless testing of the beta version in a separate app instance, enabling testing more edge cases which I personally sometimes don't even want to try to not break the existing congregation settings.

**Reply by mtdvlpr** [reply]
How about this one?
https://github.com/sircharlo/meeting-media-manager/actions/runs/12421252834

**Reply by mtdvlpr** [reply]
Or this one?

https://github.com/sircharlo/meeting-media-manager/actions/runs/12421339414

**Reply by ux-git** [reply]
@mtdvlpr unfortunately neither one worked for me. Neither as update nor as a fresh install 
<img width="372" alt="Screenshot 2024-12-19 at 23 07 16" src="https://github.com/user-attachments/assets/83df1c1c-7b4d-40b3-966e-ce3343975bd8" />

**Reply by sbscomp** [reply]
Both runs still give the error. Is the workflow notarizing the application? That's a critical step in addition to signing it.

**Reply by mtdvlpr** [reply]
> Both runs still give the error. Is the workflow notarizing the application? That's a critical step in addition to signing it.

You're right. Because they were built from a Pull Request, the notarize step was skipped. I didn't know that happened, but it makes sense. I'll re-run it manually with the notarize step.

**Reply by mtdvlpr** [reply]
I've updated the original post with 3 signed and notarized variants. Each with slightly different security configurations.

**Reply by sbscomp** [reply]
The latest beta build looks like it works!

**Reply by mtdvlpr** [reply]
Which build exactly? Without HR, with HR, or HR with JIT?

**Reply by mtdvlpr** [reply]
Please test all three of the below versions and let us know the results:

- Without HR: https://github.com/sircharlo/meeting-media-manager/actions/runs/12419841007
- With HR: https://github.com/sircharlo/meeting-media-manager/actions/runs/12427249313
- HR with JIT: https://github.com/sircharlo/meeting-media-manager/actions/runs/12427488340/job/34697375824

**Reply by ux-git** [reply]
@mtdvlpr all three versions worked for me flawlessly.

The only noticed inconvenience is regarding the website presentation feature – it opens the website main control window (not the mirrored one) full-screen as a separate desktop space, and does this NOT on the display with currently opened MMM window, but either randomly or just on another screen where the Media Player window is located. This workflow fully removes the ability to quietly open a website and demonstrate it later on 😄 And I guess the aspect ratio, at least on the JIT version, is back to having black margins on the left and right sides, but we can test in once again later.

The good thing is permissions given to one version migrate to another as expected, meaning no more weird 'Remove permission and then re-grant it' workarounds 🎉

**Reply by mtdvlpr** [reply]
Awesome! Thanks for testing. I guess we can add this to the next release then.

> it opens the website main control window (not the mirrored one) full-screen as a separate desktop space, and does this NOT on the display with currently opened MMM window, but either randomly or just on another screen where the Media Player window is located.

Does this happen in the latest released beta or only for these test versions?

**Reply by ux-git** [reply]
> Does this happen in the latest released beta or only for these test versions?

The latest beta10 has this behaviour, while latest stable version opens the website in the windowed mode as expected.

**Reply by mtdvlpr** [reply]
Continued in  #3837

**Comment by sbscomp** [comment]
I tried the without HR one I think. Can try the others later.

—
Douglas Phillips

Sent from my mobile device
________________________________

**Comment by mtdvlpr** [comment]
## Auto Updates on MacOS

Dear users,

We've just released the second version with code signing and notarizing. This means that you should now be able to experience the auto-updating feature. Please let us know if this feature works correctly. When you open your current version (24.12.2), it should give you a desktop notification saying that a new version is available and when you close the application, it should update in the background. The next time you open it, you should be on the latest version (25.1.0).

**Comment by skimark1** [comment]
Hello!

It worked perfectly on Intel Mac and Silicon Mac M2 MBA. Thanks for the hard work and diligence in making it easier to use. The update also took care of now not having to reauthorize the mirroring feature for jw.org <http://jw.org/> feature, another great timesaver. The multiple close commands is still and issue on Macs FYI.

We appreciate you brothers.

Thanks,
Mark Cillis
***@***.***

**Reply by mtdvlpr** [reply]
Thank you for the information! In regards to the close commands. Please take a look at https://github.com/sircharlo/meeting-media-manager/issues/3319#issuecomment-2478802025. That should be the behavior. In short, Cmd+Q should quit the app in 1 or 2 times depending on whether the meeting has started. If the app does not quit on first try, it should show a warning. If this is not the case, please create a new issue for it.

**Reply by mtdvlpr** [reply]
Continued in #3967

**Comment by skimark1** [comment]
I understand the warning if the meeting is running. With no meeting running, cmd-quit, the window closes, but the app is still running. Most will leave it that way, but clicking on it does nothing since it is already running with no windows showing. You have to quit it again to be able to reopen it. This creates issues between meetings, if it isn’t fully quit. I’d be glad to do a screen share with someone to show you what I’m talking about. The 3x quit from meeting running or 2x quit from no meeting running isn’t standard Mac app behavior IMHO.

Thanks,
Mark Cillis
***@***.***

**Comment by skimark1** [comment]
The warning that occurs when a meeting is running is great, but having to close it twice when no meeting is running doesn’t make sense to me. I recorded videos that will show what I think you already know. When you close it like you would Excel, the window goes away but M3 is still running. You can’t do anything with it at this point except quit again. If you don’t, it is taking system resources and most people don’t notice that it’s still running. Just a suggestion, you guys do great work!

Meeting running, 3 cmd-quit
https://github.com/user-attachments/assets/bae96d47-d6c8-4732-9b80-6fda3a27860e

Meeting not running, 2 cmd quit
https://github.com/user-attachments/assets/30d7b903-180b-4c69-8836-53a5cb17e895

Typical close of Excel, only 1 cmd quit
https://github.com/user-attachments/assets/b61edd49-d6c6-4a4e-b234-dc09adcc2f25

Thanks,
Mark Cillis

**Comment by Danreynolds124** [comment]
Hey just a heads up - Crossover 25 was released and it seems to have added
a ton of support for Windows programs on the Mac. It seems NWS will run
under this now. It may be helpful to you in porting your native version to
the Mac!

__

Dan Reynolds

815-416-9181
585-721-9181 (cell)

www.danreynoldsphotography.com

**Reply by ux-git** [reply]
@Danreynolds124 Actually, Meeting Media Manager is already available as a **native** macOS application 🙂

Unlike a Windows-first approach, Meeting Media Manager is built to work smoothly on macOS, Windows, and Linux from the day one, so users get the same features and reliable performance no matter which platform they use. Since it’s built using Electron framework, it runs directly on macOS without the need for additional compatibility layers like Crossover.

**Comment by Danreynolds124** [comment]
I’m sorry I knew this 😂. I mean to send this to NWS 😂


__

Dan Reynolds

815-416-9181
585-721-9181 (cell)

www.danreynoldsphotography.com

**Comment by mtdvlpr** [comment]
## Present Website & SL Camera Support

Dear macOS users,

In a recent update, we've added support for displaying a camera feed on the media window. This is specifically for Sign Language groups, because they need to see the signed answers from the (digital) audience.

The user who requested this feature can't get it working on his macOS device. It should work the same way as getting the Present Website feature to work, but so far I haven't been able to guide him through it. Are there any macOS users here who are willing to help him out further? If so, please open #4139 and continue the conversation there. Thank you for your willing spirits!

**Comment by wh575** [comment]
Hi friends!

I wanted to report an issue I'm having on my M1 MacBook Air, running MMM v25.7.0

Anytime I start MMM it takes quite a while to open the app. Once the app is up and running I experience lag during operation. For example, when I click the play button for a video or picture there is a few second delay. In the moment it seems like an eternity. 

On a positive note, I'm finding the current MMM version to be a bit faster than previous ones. For a while I was downgrading to v24... while it was available since because the previous v25 was almost unusable on my device.

Not sure if anyone might have some suggestions to get MMM running better on my M1. I feel like it should be much snappier than it is. 

Thanks for all your support!

https://github.com/user-attachments/assets/d8f3e8d1-c7b3-4451-abf5-eb9e2f44909a


Will

**Reply by sircharlo** [reply]
Is this true of the latest beta too?

**Reply by wh575** [reply]
Sorry I didn't have a chance to reply until now. I've just installed the latest beta (v25.7.1-beta.5)

It's similar to the previous publice release. The apps reaction time does seem to get faster after being used for a few minutes. But there is still a noticeable delay when performing various operations, playing and stopping videos. My friend has v25.7.0 installed on his M4 Macbook Air and it seems a bit sluggish as well.

For example: After hitting the play video button the first time, the media goes black for a few seconds before playing. If I hit the stop button, a few seconds will pass before the confirmation window pops up.  

Thanks so much for this app! Even though it's a bit slow right now, it's still our favorite!

Will

**Reply by sircharlo** [reply]
Ok, thanks for the feedback. I have a few ideas that I'll plug away at to see if we can find the cause.

**Comment by skimark1** [comment]
We had that happen on the Mac mini at the hall when we first installed the 25.7.0 but it was fixed by deleting and reinstalling M3. We don't run betas at the hall and I haven't tested them lately.Thanks,MarkSent from my iPhone 15 Pro MaxOn Jul 27, 2025, at 5:06 AM, Charles Olivier Savignac ***@***.***> wrote:﻿
Is this true of the latest beta too?
